here-naksha-lib-core/src/main/java/com/here/naksha/lib/core/models/storage/ContextResult.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2017-2023 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+package com.here.naksha.lib.core.models.storage;
+
+import com.here.naksha.lib.core.NakshaVersion;
+import java.util.List;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class ContextResult<FEATURE, CTX_TYPE, V_TYPE, CODEC extends FeatureCodec<FEATURE, CODEC>>
```

Comment:
Is it possible to constrain bounds for generic types: `CTX_TYPE` and `V_TYPE`?

Reply:
Discussed on call